File: server/src/redis.ts
import { createClient } from "redis";

export const redis = createClient({
  username: process.env.REDIS_USERNAME,
  password: process.env.REDIS_PW,
  socket: {
    host: process.env.REDIS_HOST!,
    port: parseInt(process.env.REDIS_PORT!),
  },
});
redis.connect();

File: engine/shared/redis.py
import redis
from shared.constants import REDIS_HOST, REDIS_PORT, REDIS_PW

def create_redis_client() -> redis.Redis:
    """Makes a new redis client object"""
    return redis.Redis(host=REDIS_HOST, port=REDIS_PORT, password=REDIS_PW if REDIS_PW else None)